librz/analysis/fcn.c

```diff
@@ -1067,6 +1101,9 @@ static int run_basic_block_analysis(RzAnalysisTaskItem *item, RzVector *tasks) {
 				bb->fail = UT64_MAX;
 			}
 			rz_analysis_task_item_new(analysis, tasks, fcn, NULL, op.jump);
+			if (continue_after_jump && is_hexagon) {
```

Comment:
Now you have `continue_after_jump`, is `is_hexagon` really necessary? I think it should not break other architectures then.

Reply:
If the `is_hexagon` flag is removed 15 tests fail. Some fail because the jump instruction should have been treated (correctly) as return statement and now it isn't. But for others I am not sure.

<details>
<summary> For example </summary>

```diff
[XX] /home/user/rizin/test/db/analysis/avr avr ATTiny RzArchProfile display
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -e asm.cpu=ATTiny48 -Qc 'aa
pd 5 @ 0x0
fs~reg
' bins/firmware/arduino_avr.bin
-- stdout
--- expected
+++ actual
@@ -1,11 +1,15 @@
-        ,=< 0x00000000  ~   jmp   entry0
-        |   ;-- PINB:
-        |   0x00000003      unaligned
-        |   ;-- DDRB:
-       ,==< 0x00000004  ~   jmp   0x1b6
-       ||   ;-- PORTB:
-       ||   0x00000005      unaligned
-       ||   ;-- PINC:
-       ||   0x00000006  ~   invalid
+|           ;-- (0x00000003) PINB:
+|       ,=< 0x00000000  ~   jmp   entry0
+|       |   ;-- DDRB:
+|       |   ;-- (0x00000005) PORTB:
+|      ,==< 0x00000004  ~   jmp   0x1b6
+|      ||   ;-- PORTC:
+|      ||   ;-- (0x00000009) PIND:
+|     ,===< 0x00000008  ~   jmp   0x1b6
+|     |||   ;-- PINA:
+|     |||   ;-- (0x0000000d) DDRA:
+|    ,====< 0x0000000c  ~   jmp   0x1b6
+|    ||||   ;-- (0x00000012) PORTCR:
+|   ,=====< 0x00000010  ~   jmp   0x1b6
    37 * registers.extended
    41 * registers.mmio

[x] Analyze all flags starting with sym. and entry0 (aa)

[XX] /home/user/rizin/test/db/analysis/avr test adding fake CPUs and loading new profiles
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'e asm.cpu=ATTiny4_fake_cpu
aa
pd 5 @ 0x03c
' bins/firmware/arduino_avr.bin
-- stdout
--- expected
+++ actual
@@ -1,8 +1,6 @@
-        ,=< 0x0000003c  ~   jmp   0x1b6
-        |   ;-- SPL:
-        |   0x0000003d      unaligned
-        |   ;-- SPH:
-        |   0x0000003e  ~   invalid
-        |   ;-- SREG:
-        |   0x0000003f      unaligned
-       ,==< 0x00000040      jmp   0x3f8
+|           ;-- (0x0000003d) SPL:
+|       ,=< 0x0000003c  ~   jmp   0x1b6
+|      ,==< 0x00000040      jmp   0x3f8
+|     ,===< 0x00000044      jmp   0x1b6
+|    ,====< 0x00000048      jmp   0x143a
+|   ,=====< 0x0000004c      jmp   0x154e

[XX] /home/user/rizin/test/db/analysis/x86_32 hang-on-aa/-aac fix (#15004)
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'e asm.bytes=true
e analysis.vars.stackname=true
aa
e asm.var=false
pdb
' bins/pe/15004.file
-- stdout
--- expected
+++ actual
@@ -1,4 +1,2 @@
-/ (loc) entry0 (int32_t arg_40h, int32_t arg_44h);
+/ (loc) entry0 (int32_t arg_44h);
 |           0x006453a1      e8d8170000     call  fcn.00646b7e
-|           0x006453a6      8d642444       lea   esp, [arg_44h]
-|       ,=< 0x006453aa      0f85a6300000   jne   0x648456

[XX] /home/user/rizin/test/db/analysis/x86_64 noreturn of reloc-functions
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'aaa
afbr 0x80b0
' bins/elf/ls
-- stdout
--- expected
+++ actual
@@ --1,4 +-1,3 @@
 0x000081cb
-0x00008237

[XX] /home/user/rizin/test/db/analysis/mach0 aav find xrefs
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'e analysis.loads=true
e analysis.in=range
e analysis.from=0x8000
e analysis.to=0x8000+4K
e asm.bytes=true
aac
afna
aar
aan
aav
# aa;afna;aac;aar;aan;aav
pd 17 @ 0x8058
' bins/mach0/macho-bash
-- stdout
--- expected
+++ actual
@@ --1,34 +-1,34 @@
 |       `=< 0x00008058      03e4           b     0x7862
-            0x0000805a      0000           movs  r0, r0
-            ; DATA XREF from sub.free_73a0 @ 0x7d30
-            0x0000805c      .dword 0x000693da
-            ; DATA XREF from sub.free_73a0 @ 0x7d3e
-            0x00008060      .dword 0x0006c1f4
-            ; DATA XREF from sub.free_73a0 @ 0x7d48
-            0x00008064      .dword 0x0006c1a6
-            ; DATA XREF from sub.free_73a0 @ 0x7e30
-            0x00008068      .dword 0x0006c102
-            ; DATA XREF from sub.free_73a0 @ 0x7e3a
-            0x0000806c      .dword 0x0006c0b4
-            ; DATA XREF from sub.free_73a0 @ 0x7e58
-            0x00008070      .dword 0x0006c0da
-            ; DATA XREF from sub.free_73a0 @ 0x7e62
-            0x00008074      .dword 0x0006c08c
-            ; DATA XREF from sub.strcpy_7890 @ 0x7e80
-            0x00008078      .dword 0x0006c0b2
-            ; DATA XREF from sub.strcpy_7890 @ 0x7e8a
-            0x0000807c      .dword 0x0006c064
-            ; DATA XREF from sub.strcpy_7890 @ 0x7ed8
-            0x00008080      .dword 0x00069232
-            ; DATA XREF from sub.strcpy_7890 @ 0x7ee8
-            0x00008084      .dword 0x0006c04a
-            ; DATA XREF from sub.strcpy_7890 @ 0x7ef2
-            0x00008088      .dword 0x0006bffc
-            ; DATA XREF from sub.free_73a0 @ 0x7f64
-            0x0000808c      .dword 0x0006bfce
-            ; DATA XREF from sub.free_73a0 @ 0x7f72
-            0x00008090      .dword 0x0006bf7c
-            ; DATA XREF from sub.free_73a0 @ 0x7fcc
-            0x00008094      .dword 0x0006bf66
+|           0x0000805a      0000           movs  r0, r0
+|           ; DATA XREF from sub.free_73a0 @ 0x7d30
+|           0x0000805c      .dword 0x000693da
+|           ; DATA XREF from sub.free_73a0 @ 0x7d3e
+|           0x00008060      .dword 0x0006c1f4
+|           ; DATA XREF from sub.free_73a0 @ 0x7d48
+|           0x00008064      .dword 0x0006c1a6
+|           ; DATA XREF from sub.free_73a0 @ 0x7e30
+|           0x00008068      .dword 0x0006c102
+|           ; DATA XREF from sub.free_73a0 @ 0x7e3a
+|           0x0000806c      .dword 0x0006c0b4
+|           ; DATA XREF from sub.free_73a0 @ 0x7e58
+|           0x00008070      .dword 0x0006c0da
+|           ; DATA XREF from sub.free_73a0 @ 0x7e62
+|           0x00008074      .dword 0x0006c08c
+|           ; DATA XREF from sub.free_73a0 @ 0x7e80   <--------- free instead of strcpy
+|           0x00008078      .dword 0x0006c0b2
+|           ; DATA XREF from sub.free_73a0 @ 0x7e8a   <--------- free instead of strcpy
+|           0x0000807c      .dword 0x0006c064
+|           ; DATA XREF from sub.free_73a0 @ 0x7ed8   <--------- free instead of strcpy
+|           0x00008080      .dword 0x00069232
+|           ; DATA XREF from sub.free_73a0 @ 0x7ee8   <--------- free instead of strcpy
+|           0x00008084      .dword 0x0006c04a
+|           ; DATA XREF from sub.free_73a0 @ 0x7ef2   <--------- free instead of strcpy
+|           0x00008088      .dword 0x0006bffc
+|           ; DATA XREF from sub.free_73a0 @ 0x7f64
+|           0x0000808c      .dword 0x0006bfce
+|           ; DATA XREF from sub.free_73a0 @ 0x7f72
+|           0x00008090      .dword 0x0006bf7c
+|           ; DATA XREF from sub.free_73a0 @ 0x7fcc
+|           0x00008094      .dword 0x0006bf66

```

</details>

Reply:
OK.